### PR TITLE
[Merged by Bors] - chore(SetTheory/Cardinal/Ordinal): split file

### PR DESCRIPTION
**Pull request by vihdzp**
We split this file into two:

- `Cardinal/Aleph`: contains the definitions for the aleph and beth functions (and soon, the omega function).
- `Cardinal/Arithmetic`: any results on cardinal arithmetic that depend on `c * c = c` (which requires some basic properties of ordinals to be proven)

These files are otherwise unchanged, with the exception of tweaking some section headers.

---

- [x] depends on: #17480

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Review by YaelDillies — COMMENTED** [review]
Can you add some `assert_not_exists` demonstrating the import sanitisation?

**Comment by vihdzp** [comment]
I added `assert_not_exists Cardinal.mul_eq_self` on `Aleph` to ensure none of the arithmetical results depending on it have been defined at that point. Not sure if this is what you meant.

**Comment by YaelDillies** [comment]
Yes, that's what I meant!

maintainer merge

File: Mathlib/SetTheory/Cardinal/Arithmetic.lean [hunk 1]
```diff
@@ -642,10 +229,7 @@ theorem mul_eq_left_iff {a b : Cardinal} : a * b = a ↔ max ℵ₀ b ≤ a ∧
     · rw [mul_eq_max_of_aleph0_le_left ha hb, max_eq_left hab]
     all_goals simp
 
-end mul
-
 /-! ### Properties of `add` -/
-section add
```
Comment (b-mehta) [inline Mathlib/SetTheory/Cardinal/Arithmetic.lean:648]: What's the purpose of removing all the `section`s throughout this file? They're often valuable, for instance navigating a file using Explorer view in VSCode.

Comment (vihdzp) [reply Mathlib/SetTheory/Cardinal/Arithmetic.lean:648]: I felt like they were made redundant by the section headers. I don't think the pattern of
```
/-! ### Foo -/
section foo

end foo
```
is all that common (and imo if you're not introducing variables or anything it's just more bloat).

I'll introduce them back if you want though.

Comment (YaelDillies) [reply Mathlib/SetTheory/Cardinal/Arithmetic.lean:648]: > I don't think the pattern of
> 
> ```
> /-! ### Foo -/
> section foo
> 
> end foo
> ```
> 
> is all that common

I actually think this is *the* pattern?

Comment (vihdzp) [reply Mathlib/SetTheory/Cardinal/Arithmetic.lean]: Well, I've added the sections back.

**Comment by b-mehta** [comment]
Thanks!

bors merge

**Pull request closed without merging**